src/store/adjustVault.ts

```diff
@@ -87,9 +87,9 @@ export const vaultToAdjustAtom = atom<VaultToAdjust | null>(get => {
   };
 });
 
-export const collateralInputValueAtom = atom<bigint | null>(null);
+export const collateralInputValueAtom = atom<NatValue | null>(null);
```

Comment:
good for documentation, though may be deceptive since `NatValue = bigint`. If you had a `useNatAtom` you could validate it.

Reply:
It's tough, but I just gave the atom a setter which does the validation, hopefully that helps